lib/browser/api/web-contents.ts

```diff
@@ -533,6 +533,19 @@ WebContents.prototype._init = function () {
     enumerable: true
   });
 
+  // Add navigationHistory property which handles session history,
+  // maintaining a list of navigation entries for backward and forward navigation.
+  Object.defineProperty(this, 'navigationHistory', {
+    get () {
+      return {
+        getActiveIndex: this._getActiveIndex.bind(this),
+        length: this._length.bind(this),
+        getEntryAtIndex: this._getNavigationEntryAtIndex.bind(this)
+      };
+    },
```

Comment:
I think this will result in a unique object being created on each call to the getter.  i.e. `webContents.navigationHistory !== webContents.navigationHistory` which is fairly weird IMO.

This should probably just be a `value` set (if this was inspired by the `ipc` thing above, that one is also weird, sorry 😆)

```suggestion
    value: {
      getActiveIndex: this._getActiveIndex.bind(this),
      length: this._length.bind(this),
      getEntryAtIndex: this._getNavigationEntryAtIndex.bind(this)
    },
```

Reply:
Nice catch! yeah I tested this and you are right. Changed it to value.